### PR TITLE
helper.viewers dict to better generalize reference vs id

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request fixes the keys in `imviz.viewers` for additional viewers to fallback on the id when reference does not exist so the keys are not `None`, but rather `imviz-2`, etc.  

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Any chance a test can be added to https://github.com/spacetelescope/jdaviz/blob/main/jdaviz/configs/imviz/tests/test_viewers.py ?

Especially given that the prev patch obviously didn't do good enough, requiring this patch, which means a test should have been added in the first place. 

Thanks!

File: jdaviz/configs/default/plugins/viewers.py
Comment (cshanahan1) [inline jdaviz/configs/default/plugins/viewers.py:321]: this can be shortened (but also nbd if you want to keep it as is)
return self.reference_id if self.reference is None else self.reference

**Review by cshanahan1 — APPROVED** [review]
lgtm, one small suggestion but not crucial

File: jdaviz/configs/default/plugins/viewers.py
Comment (pllim) [reply jdaviz/configs/default/plugins/viewers.py:321]: You mean `self.reference_id if self.reference is not None else self.reference` ?

Comment (kecnry) [reply jdaviz/configs/default/plugins/viewers.py:321]: I avoided that since `self.reference` isn't cached and requires looping over the viewer-store and I didn't want to have to call that twice.  Is it worth implementing caching so that we can do this?

**Comment by kecnry** [comment]
@pllim - regression test added

File: jdaviz/configs/default/plugins/viewers.py
Comment (pllim) [reply jdaviz/configs/default/plugins/viewers.py:321]: Naw.

Comment (pllim) [reply jdaviz/configs/default/plugins/viewers.py:321]: Might want to add a note about this so future dev won't try to shorten it?

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

File: jdaviz/configs/default/plugins/viewers.py
Comment (pllim) [reply jdaviz/configs/default/plugins/viewers.py:321]: I didn't merge because I don't know if you want to follow up on this, so I'll let you merge when you think this is ready. You already have 2 approvals.

**Pull request merged**
